assessmentModel/src/commonMain/kotlin/AsyncActionConfiguration.kt

```diff
@@ -0,0 +1,130 @@
+package org.sagebionetworks.assessmentmodel
+
+/**
+ * [AsyncActionConfiguration] defines general configuration for an asynchronous action that should be run in
+ * the background. Depending upon the parameters and how the action is set up, this could be something that is run
+ * continuously or else is paused or reset based on a timeout interval.
+ */
+interface AsyncActionConfiguration : ResultMapElement {
+
+    /**
+     * An identifier marking the step to start the action. If `null`, then the action will be started when the
+     * [AsyncActionContainer] is started.
+     */
+    val startStepIdentifier: String?
+}
+
+/**
+ * [RecorderConfiguration] is used to configure a recorder. For example, recording accelerometer data or video.
+ */
+interface RecorderConfiguration : AsyncActionConfiguration {
+
+    /**
+     * A localized string (or localization key) describing the reason for using this recorder.
+     */
+    val reason: String?
+
+    /**
+     * Is this recorder required to run the assessment or is it optional? Some active tasks can run and will return
+     * valuable results without the augmented results from a sensor. For other assessments, the recorder is required
+     * in order for the measurement of the assessment to be meaningful.
+     */
+    val optional: Boolean
+
+    /**
+     * Most recorders will require one or more permissions.
+     */
+    val permissions: List<PermissionInfo>?
+
+    /**
+     * An identifier marking the step at which to stop the action. If `null`, then the action will be stopped when the
+     * task is stopped.
+     */
+    val stopStepIdentifier: String?
+
+    /**
+     * Whether or not the recorder must be allowed to run when the app is in the background. Default = `false`.
+     */
+    val requiresBackground: Boolean
+        get() = false
+
+    /**
+     * Should the file used in a previous run of a recording be deleted or should a unique file name be used to
+     * identify the file when a step is repeated? Default = `true`.
+     */
+    val shouldDeletePrevious: Boolean
+        get() = true
+
+    /**
+     * Set the flag to `true` to encode the samples as a CSV file. Default = `false`.
+     */
+    val usesCSVEncoding : Boolean
```

Comment:
Are all recorders going to have the ability to encode samples as a CSV file? Wondering if this should be in a separate interface or just on the implementation classes.

Reply:
Well, they all do in SageResearch b/c it's easy but yeah, good point.